transit_odp/fares_validator/tests/test_mandatory_type_of_frame_refs.py

```diff
@@ -0,0 +1,374 @@
+import pytest
+from lxml import etree
+
+from transit_odp.fares_validator.views.functions import (
+    check_fare_frame_type_of_frame_ref_present_fare_price,
+    check_fare_frame_type_of_frame_ref_present_fare_product,
+    check_resource_frame_type_of_frame_ref_present,
+)
+
+NAMESPACE = {"x": "http://www.netex.org.uk/netex"}
+RESOURCE_FRAME_X_PATH = "//x:dataObjects/x:CompositeFrame"
+FARE_FRAME_X_PATH = "//x:dataObjects/x:CompositeFrame/x:frames/x:FareFrame"
+
+
+def get_lxml_element(xpath, string_xml):
+    doc = etree.fromstring(string_xml)
+    elements = doc.xpath(xpath, namespaces=NAMESPACE)
+    return elements
+
+
+@pytest.mark.parametrize(
+    (
+        "fare_frame",
+        "fare_frame_valid_id",
+        "type_of_frame_ref",
+        "type_of_frame_ref_ref_present",
+        "type_of_frame_ref_ref_valid",
+        "expected",
+    ),
+    [
+        (True, True, True, True, True, None),
+        (True, False, False, False, False, None),
+        (
+            True,
+            True,
+            False,
+            False,
+            False,
+            [
+                "violation",
+                "7",
+                "Mandatory element 'TypeOfFrameRef' is missing from 'FareFrame' - UK_PI_FARE_PRICE",
+            ],
+        ),
+        (True, True, True, False, False, ""),
+        (
+            True,
+            True,
+            True,
+            True,
+            False,
+            [
+                "violation",
+                "8",
+                "Attribute 'ref' of element 'TypeOfFrameRef' does not contain 'UK_PI_FARE_PRICE'",
```

Comment:
[1]:This error message does not match one defined in validation_message.py. There is a small difference in wording.